scan/git.go

```diff
@@ -26,6 +26,16 @@ type gitRepo struct {
 	subdir string
 }
 
+// IsGitRepo checks whether current directory is a GIT directory
+func (s *Scanner) IsGitRepo(ctx context.Context, cmdDir string) bool {
```

Comment:
IMO remove this code path and just pass in the `--branch` and `--commit` to avoid running the commands 2x. If those are specified with a local context you can populate the dependencies from it

Reply:
I think that'll ensure we're consistent with the branch and commit rendering since they can muck with the context at this point, checking out other branches, etc.

Reply:
This is done to also support passing in remote context.

Like `acb build https://github.com/blah.git`

cc @northtyphoon

Reply:
I know... let me if the comment doesn't make sense and I can rephrase it.

Reply:
If we want to completely avoid calling the commands 2x, pass --commit all the way down to the Scanner.Scan to fill GitHeadRev. If we still want to support `acb build https://github.com/blah.git`, we will likely need a new ObtainSourceCode method to do something like the following:
```
retrieve the source code
if commit is provided, return commit
else try to git rev-parse to get the commit
```

Reply:
i dont see why calling it 2x would be a problem..

Reply:
its like: `download` -> ObtainSourceCode (1st time, remote ctx) -> get commitID (inside workDir) -> pass it to builder

RunStep -> Scan -> ObtainSourceCode (2nd time, local ctx) -> get commitID (inside workDir)